### PR TITLE
Fix `manual_map` suggestion for `if let.. else ... if let.. else` chain

### PR DESCRIPTION
**Pull request by Jarcho**
fixes: #6847

changelog: Fix `manual_map` suggestion for `if let.. else ... if let.. else` chain

### CODE REVIEW

File: clippy_lints/src/manual_map.rs [hunk 1]
```diff
@@ -182,13 +185,36 @@ impl LateLintPass<'_> for ManualMap {
...
+                } else {
+                    format!("{}{}.map({})", scrutinee_str, as_ref_str, body_str)
+                },
                 app,
             );
         }
     }
 }
 
+fn is_else_of_if_let_else(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
```
Comment (Manishearth) [inline clippy_lints/src/manual_map.rs]: move to utils (with docs)?

(Probably with all the other desugar functions)

**Review by Manishearth — APPROVED** [review]
Approach seems good, the util function should be in utils and documented though

File: clippy_lints/src/manual_map.rs (hunk 1 above)
Comment (Jarcho) [reply clippy_lints/src/manual_map.rs]: That was fixed as you were writing that. Forgot to commit that change.

**Pull request merged**